### PR TITLE
Feature/bc 425 sign up and email verification flow

### PR DESCRIPTION
**Pull request by yeezick**
- refactor sign up screen with MUI inputs 
  - complete with input validations
- add user state to redux and token to localstorage after user verifies email 
- renames utilities to utils 
- separates interfaces to its own directory
- disable strict mode and noUnusedVars in tsconfig for better dev experience and velocity

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,4 +1,9 @@
+body {
+  background-color: #fff;
+}
+
 .App {
+  height: 100vh;
```
Comment (yeezick) [inline src/App.css:6]: jason handled this differently, prefer his method once it's merged

File: src/__tests__/screens/SignUp.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  test('User can fill out the sign up form', async () => {
+    const { user } = render(<SignUp />, initialState)
+    // const [
+    //   firstNameInput,
+    //   lastNameInput,
+    //   emailInput,
+    //   // passwordInput,
+    //   // confirmPasswordInput,
+    // ] = screen.getAllByRole('textbox')
+    const firstNameInput = screen.getByLabelText(/first name/i)
```
Comment (yeezick) [inline src/__tests__/screens/SignUp.spec.tsx:27]: ignore, tests need jesus

File: src/components/Inputs/inputs.scss [hunk 3]
```diff
@@ -0,0 +1,7 @@
+.error {
+  color: red;
+}
+
+.warning {
+  color: yellow
+}
```
Comment (yeezick) [inline src/components/Inputs/inputs.scss:7]: delete unused file

File: src/layout/Layout.scss [hunk 4]
```diff
@@ -1,7 +1,11 @@
...
-  margin: 50px 272px;
-  transition: all 0.1s;
-  width: calc(100% - 272px);
+.layout-container {
+  margin-top: 73px;
+  padding-top: 20px;
+
+  .active {
+    transition: all 0.1s;
+    // width: calc(100% - 272px);
```
Comment (yeezick) [inline src/layout/Layout.scss:7]: delete commented out code

File: src/screens/Auth/SignUp/SignUp.scss [hunk 5]
```diff
@@ -45,17 +45,17 @@
   padding: 32px;
   width: 320px;
 
-  button {
-    @include form-buttons();
-  }
+  // button {
+  //   @include form-buttons();
+  // }
```
Comment (yeezick) [inline src/screens/Auth/SignUp/SignUp.scss:50]: delete comments

File: src/utils/data/notificationConstants.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import {
   NotificationInterface,
   NotificationState,
-} from '../types/NotificationInterface'
+} from '../../interfaces/NotificationInterface'
```
Comment (yeezick) [inline src/utils/data/notificationConstants.ts:4]: use alias import

File: src/utils/data/userConstants.ts [hunk 7]
```diff
@@ -2,7 +2,7 @@ import {
   SignUpInterface,
   UiSliceInterface,
   UserInterface,
-} from '../types/UserInterface'
+} from '../../interfaces/UserInterface'
```
Comment (yeezick) [inline src/utils/data/userConstants.ts:5]: alias import

File: src/App.css (hunk 1 above)
Comment (yeezick) [reply src/App.css:6]: reviewer: please resolve as this is a sidenote more than a real comment

File: src/__tests__/screens/SignUp.spec.tsx (hunk 2 above)
Comment (yeezick) [reply src/__tests__/screens/SignUp.spec.tsx:27]: reviewer: please resolve as this is a sidenote more than a real comment

File: src/screens/Auth/SignUp/SignUp.scss (hunk 5 above)
Comment (gooliajulia) [reply src/screens/Auth/SignUp/SignUp.scss:50]: are you telling yourself to delete these?

**Review by gooliajulia — APPROVED** [review]

File: src/App.css [hunk 8]
```diff
@@ -1,4 +1,9 @@
...
   text-align: center;
 }
```
Comment (yeezick) [inline src/App.css:9]: undo changes

File: src/layout/Footer/Footer.scss [hunk 9]
```diff
@@ -5,9 +5,7 @@
   color: #ffffff;
   display: flex;
   flex-direction: column;
-  height: auto;
-  left: 0;
-  margin-top: 70px;
+  height: fit-content;
   padding: 20px 20px 0 20px;
```
Comment (yeezick) [inline src/layout/Footer/Footer.scss:9]: undo

**Comment by gooliajulia** [comment]
this latest commit is just reverting some styling bc of overlap with Jason's work?

**Comment by yeezick** [comment]
> this latest commit is just reverting some styling bc of overlap with Jason's work?

That's correct

**Review by gooliajulia — APPROVED** [review]

**Pull request merged**